src/utils/messaging/strategies/iframe/iframe-messaging.strategy.ts

```diff
@@ -0,0 +1,311 @@
+import type { ProtocolMap } from "@arconnect/webext-bridge";
+import { nanoid } from "nanoid";
+import type { ApiCall, ApiResponse } from "shim";
+import {
+  EMBEDDED_ANCESTOR_TAB_ID,
+  EMBEDDED_IFRAME_TAB_ID
+} from "~utils/embedded/embedded.constants";
+import { getEmbeddedAncestorOrigin } from "~utils/embedded/embedded.utils";
+import { isInsideIframe } from "~utils/embedded/iframe.utils";
+import { log, LOG_GROUP } from "~utils/log/log.utils";
+import type {
+  MessageData,
+  MessageID,
+  OnMessageCallback
+} from "~utils/messaging/messaging.types";
+
+const messageHandlersByMessageID: Partial<
+  Record<MessageID, Set<OnMessageCallback<MessageID>>>
+> = {};
+
+let targetIframe: HTMLIFrameElement = null;
+let targetOrigin = "";
+
+export function setEmbeddedTargetIframe(iframeElement: HTMLIFrameElement) {
+  targetIframe = iframeElement;
+  targetOrigin = new URL(iframeElement.src).origin;
+}
+
+let messageCounter = 0;
+
+function getPostMessageFunction<K extends MessageID>(
+  messageData: MessageData<K>
+): () => Promise<ReturnType<OnMessageCallback<K>>> {
+  let postMessageTargetOrigin = "";
+
+  const { destination, messageId, data } = messageData;
+
+  // TODO: isInsideIframe is an incorrect check because we might be running the app standalone.
```

Comment:
See here. The app won't run standalone once this is merged until this is addressed.